### PR TITLE
304 refactor check init

### LINKED ISSUE #304: Refactor the dependency and init/execute flow for Checks

### PR DESCRIPTION
**Pull request by spaudanjo**
fixes #304

### CODE REVIEW

**Comment by spaudanjo** [comment]
👎

**Comment by spaudanjo** [comment]
Little cleanup WIP

File: server/checks/Approval.js [hunk 1]
```diff
@@ -16,6 +16,20 @@ export default class Approval extends Check {
   static NAME = 'Approval check'
   static HOOK_EVENTS = [EVENTS.PULL_REQUEST, EVENTS.ISSUE_COMMENT]
 
+  constructor(github, pullRequestHandler) {
```
Comment (mfellner) [inline server/checks/Approval.js]: Let's also add some JSDoc to indicate the type, e.g.
```javascript
/**
 * @param {GithubService} github
 * @param {PullRequestHandler} pullRequestHandler
 */
constructor(github, pullRequestHandler) {
```

File: server/handler/HookHandler.js [hunk 2]
```diff
@@ -3,16 +3,26 @@ import merge from 'lodash/merge'
...
+              repositoryHandler = defaultRepositoryHandler,
+              pullRequestHandler = defaultPullRequestHandler) {
+    this.githubService = githubService
+    this.repositoryHandler = repositoryHandler
+
+    this.pullRequestHandler = pullRequestHandler
+    this.approval = new Approval(this.githubService, this.pullRequestHandler)
+
+    this.autobranch = new Autobranch(this.githubService)
+
```
Comment (mfellner) [inline server/handler/HookHandler.js]: Let's not use any blank lines between this initialisations.

File: test/server/approval.test.js [hunk 3]
```diff
@@ -73,7 +73,7 @@ describe('Approval#countApprovalsAndVetos', () => {
       body: ':+1:' // is ignored because mfellner already approved
     }]
     try {
-      const {approvals} = await Approval.countApprovalsAndVetos(null, DEFAULT_REPO, comments, DEFAULT_CONFIG.approvals, null)
+      const {approvals} = await (new Approval(null, null)).countApprovalsAndVetos(DEFAULT_REPO, comments, DEFAULT_CONFIG.approvals)
```
Comment (mfellner) [inline test/server/approval.test.js]: Let's instantiate the Approval with it's own variable on the line above to make things a bit cleaner.

File: test/server/approval.test.js [hunk 4]
```diff
@@ -83,7 +83,7 @@ describe('Approval#countApprovalsAndVetos', () => {
 })
 
 describe('Approval#getCommentStatsForConfig', () => {
-  var github
+  var github, approval
```
Comment (mfellner) [inline test/server/approval.test.js]: This is a good opportunity to use a `let` here instead.

File: test/server/commitmessage.test.js [hunk 5]
```diff
@@ -52,13 +52,14 @@ describe('CommitMessage', () => {
     })
   })
   describe('#execute', () => {
-    var github
+    var github, commitMessage
```
Comment (mfellner) [inline test/server/commitmessage.test.js]: Let's use a let!

File: test/server/autobranch.test.js [hunk 6]
```diff
@@ -64,18 +64,19 @@ describe('Autobranch', () => {
     })
   })
   describe('#execute', () => {
-    var github
+    var github, autobranch
```
Comment (mfellner) [inline test/server/autobranch.test.js]: Let's use a let!

**Comment by mfellner** [comment]
👍

**Comment by mfellner** [comment]
👍

**Pull request merged**